rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/TEContentProvider.xtend

```diff
@@ -62,6 +58,21 @@ class TEContentProvider implements ITreeContentProvider {
 		return null
 	}
 
+	/**
+	 * returns the parent folder as ClasspathEntry or null if it is not a ClasspathEntry.
+	 */
+	def IClasspathEntry getParentClassPathOf(IResource element) {
+		if (element.project != null) {
+			val list = getChildren(element.project).filter(IClasspathEntry).filter [
+				workspaceRootHelper.getRoot.getFolder(it.path).members.contains(element)
+			]
+			if (!list.empty) {
```

Comment:
No need to check if the list is empty, `list.head` will return null if the list is empty.